Dialogs/PreferencesDialog.cpp

```diff
@@ -1,22 +1,88 @@
 #include "PreferencesDialog.h"
 #include "ui_PreferencesDialog.h"
 
+#include <QPushButton>
+#include <QSettings>
 #include <QStyleFactory>
 
 PreferencesDialog::PreferencesDialog(QWidget *parent) : QDialog(parent), ui(new Ui::PreferencesDialog) {
   ui->setupUi(this);
   this->setWindowFlag(Qt::WindowContextHelpButtonHint, false);
 
-  foreach (QString styleName, QStyleFactory::keys()) {
-    ui->styleCombo->addItem(styleName);
-    if (style()->objectName().toLower() == styleName.toLower())
-      ui->styleCombo->setCurrentIndex(ui->styleCombo->count() - 1);
-  }
+  foreach (QString styleName, QStyleFactory::keys()) { ui->styleCombo->addItem(styleName); }
+
+  this->reset();
+
+  connect(ui->buttonBox->button(QDialogButtonBox::RestoreDefaults), &QAbstractButton::clicked, this,
+          &PreferencesDialog::restoreDefaultsClicked);
+  connect(ui->buttonBox->button(QDialogButtonBox::Apply), &QAbstractButton::clicked, this,
+          &PreferencesDialog::applyClicked);
 }
 
 PreferencesDialog::~PreferencesDialog() { delete ui; }
 
-void PreferencesDialog::on_styleCombo_activated(const QString &styleName) {
+static inline QString preferencesKey() { return QStringLiteral("Preferences"); }
+static inline QString generalKey() { return QStringLiteral("General"); }
+static inline QString appearanceKey() { return QStringLiteral("Appearance"); }
+
+void PreferencesDialog::apply() {
+  QSettings settings;
+  settings.beginGroup(preferencesKey());
+
+  settings.beginGroup(generalKey());
+  settings.setValue("documentationURI", ui->documentationLineEdit->text());

```

Comment:
@JoshDreamland do you think I should make constants for all of these keys to avoid mistakes/typos causing them to not work? What's a good clean way to do that? I was already doing it for the groups (e.g, appearanceKey()). Is that acceptable?

Reply:
I decided to do it anyway: f23ade6188e5314ba8961a15c68e61cea1772f3d